### PR TITLE
feat(branch_protection): add blocks_creations attribute

### PR DESCRIPTION
**Pull request by james-callahan**
Fixes https://github.com/integrations/terraform-provider-github/issues/1172

### CODE REVIEW

**Review by joshdk — APPROVED** [review]
This would be really nice to have, and the provider changes are fairly rote which is nice.

File: go.mod [hunk 1]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/hashicorp/hcl/v2 v2.3.0 // indirect
 	github.com/hashicorp/terraform-config-inspect v0.0.0-20191212124732-c6ae6269b9d7 // indirect
 	github.com/hashicorp/terraform-plugin-sdk v1.7.0
-	github.com/shurcooL/githubv4 v0.0.0-20220106005112-0707a5a90543
+	github.com/shurcooL/githubv4 v0.0.0-20220520033151-0b4e3294ff00
```
[the remark this answers is not in the document]
Comment (james-callahan) [reply go.mod:13]: Yes. The `BlocksCreations` field was only added in a recent update of the `githubv4` module.

**Comment by james-callahan** [comment]
Rebased to fix merge conflicts.
@kfcampbell could you have a look?

**Comment by maxexcloo** [comment]
Will this be merged?

**Comment by james-callahan** [comment]
Could someone please have a look at this PR?

**Review by kfcampbell — APPROVED** [review]

**Comment by kfcampbell** [comment]
Apologies for the delay, I'll get this merged.

**Pull request merged**